### ISSUE TITLE
Dynamics popup refinements (must be done for 4.5)

### ISSUE THREAD

**Issue by avvvvve** [musescore/MuseScore#26018]
Refinements to https://github.com/musescore/MuseScore/pull/24125

See issues 1, 2, 3, and 6 from my comment [here](https://github.com/musescore/MuseScore/pull/24125#issuecomment-2307610380) (pasted below, but there's some discussion at the link)

_____

1. Users must be able to navigate to/within the popup via keyboard navigation, i.e. with a dynamic selected, "Tab" should move focus to the left arrow, then arrow keys can be used to navigate. Pressing enter/spacebar acts as clicking. See the "Capo" and "Harp pedal diagram" elements for examples of how this is already implemented!

2. Certain keyboard interactions on dynamics in the score no longer work with the popup open. Left/right nudging (with and without shift held) is not possible. Here's how I would expect this to work:
     - Left/right/up/down: When you use an arrow key, the dynamic should move and the popup should be dismissed
     - Shift: When shift is held down, the time grid (pictured below) should appear on the staff and the popup should be dismissed. Then, the user should be able to use left/right to nudge it to time positions. This was being developed for 4.4 concurrently with your project, so no worries if you didn't see this behavior before.
          <img width="288" alt="image" src="https://github.com/user-attachments/assets/bb3e0ee1-5c8c-4e57-b65b-fdcde975b2d1">

3. 'Esc' is not working correctly. This is in part due to a regression to 4.3.2 _not_ caused by this PR, but even once that is fixed we should include this additional behavior: When a dynamic is selected and the popup is open, one 'esc' should dismiss the popup but leave the dynamic selected, and a second 'esc' should deselect the dynamic.

6. The order of the "ppppp pppp ppp" and the crescendo/decrescendo pages should be swapped, so when you move left from the first page, you get to the "ppppp ..." page. Left again would take you to cres/dec. (Oversight on my part, updated figma with the correct order)
     <img width="200" alt="image" src="https://github.com/user-attachments/assets/8a91f2ec-3b69-4cab-a595-1334006c7060">

**Comment by bkunda** [comment]
Point 2 (concerning the SHIFT key to move a dynamic to a different time position) is particularly important. I'm actually finding this to be a significant issue in real use – particularly since we were until recently able to do this!

**Comment by Eism** [comment]
@bkunda @avvvvve 
Fixed in #26189 and #26044 but with some modifications:

1. Instead, when they reach the edge and try to navigate further, I switch the page, so we don't allow the user to navigate to the arrows. 

2. and 3. In general, the popup doesn't receive focus and the user doesn't need to care about it, it doesn't interfere with the interaction with the element. Therefore, during any interaction with the element, the popup remains visible. When you press Esc, the interaction will occur on the element, not on the popup, and it will work as a deselect. But if you go to the popup via Tab, then all interaction will occur on the popup - when you press Esc, the popup will close and the next press of Esc will remove the selection since the interaction has moved from the popup to the element.

please test #26189
